lib/foreman_inventory_upload/async/generate_all_reports_job.rb

```diff
@@ -16,17 +18,19 @@ def perform
           total_hosts = ForemanInventoryUpload::Generators::Queries.for_org(organization.id, use_batches: false).count
 
           if total_hosts <= ForemanInventoryUpload.max_org_size
-            GenerateReportJob.perform_later(ForemanInventoryUpload.generated_reports_folder, organization.id)
+            plan_generate_report(ForemanInventoryUpload.generated_reports_folder, organization)
           else
             logger.info("Skipping automatic uploads for organization #{organization.name}, too many hosts (#{total_hosts}/#{ForemanInventoryUpload.max_org_size})")
           end
         end.compact
-      ensure
-        self.class.set(:wait => 24.hours).perform_later
       end
 
-      def self.singleton_job_name
-        name
+      def rescue_strategy_for_self
+        Dynflow::Action::Rescue::Fail
+      end
+
+      def plan_generate_report(folder, organization)
+        ForemanTasks.async_task(ForemanInventoryUpload::Async::GenerateReportJob, folder, organization.id)
```

Comment:
Is there a reason for going with a separate execution plan for each organization instead of having this just as actions with different inputs within the same execution plan?